### ISSUE TITLE
pycoast compatability issue with pyproj 2+

### ISSUE THREAD

**Issue by ColinDuff** [pytroll/pycoast#29]
when running the following on a resampled scene 
scn.show('overview', overlay={'coast_dir': ' /tcenas/proj/mpef/pytroll_training/data/shapes/', 'color': (255, 0, 0), 'resolution': 'i'}) i got the following error

``` "/tcenas/home/cduff/miniconda3/envs/satpyEnv/lib/python3.7/site-packages/satpy/scene.py", line 1145, in show
    img = get_enhanced_image(self[dataset_id].squeeze(), overlay=overlay)
  File "/tcenas/home/cduff/miniconda3/envs/satpyEnv/lib/python3.7/site-packages/satpy/writers/__init__.py", line 450, in get_enhanced_image
    img = add_overlay(img, dataset.attrs['area'], fill_value=fill_value, **overlay)
  File "/tcenas/home/cduff/miniconda3/envs/satpyEnv/lib/python3.7/site-packages/satpy/writers/__init__.py", line 267, in add_overlay
    resolution=resolution, width=width, level=level_coast)
  File "/tcenas/home/cduff/miniconda3/envs/satpyEnv/lib/python3.7/site-packages/pycoast/cw_agg.py", line 414, in add_coastlines
    y_offset=y_offset)
  File "/tcenas/home/cduff/miniconda3/envs/satpyEnv/lib/python3.7/site-packages/pycoast/cw_base.py", line 591, in _add_feature
    x_offset=x_offset, y_offset=y_offset, **kwargs)
  File "/tcenas/home/cduff/miniconda3/envs/satpyEnv/lib/python3.7/site-packages/pycoast/cw_base.py", line 518, in add_shapes
    lon_min, lon_max, lat_min, lat_max = _get_lon_lat_bounding_box(area_extent, x_size, y_size, prj)
  File "/tcenas/home/cduff/miniconda3/envs/satpyEnv/lib/python3.7/site-packages/pycoast/cw_base.py", line 880, in _get_lon_lat_bounding_box
    if prj.is_latlong():
AttributeError: 'Proj' object has no attribute 'is_latlong' ```

i was using the latest version of satpy 0.16.1 and pypropj 2.2.1.

Slack message was posted and a suggestion to downgrade to pyproj 1+ as David Hoese believes there is a compatability issue with pycost and pyproj2

**Comment by ColinDuff** [comment]
i noticed this from pyproj github issues page

```def is_latlong(self): 
         """ 
         Returns 
         ------- 
         bool: True if projection in geographic (lon/lat) coordinates. 
         """ 
         warnings.warn("'is_latlong()' is deprecated. Please use 'crs.is_geographic'.") 
         return self.crs.is_geographic  ```

This warning was removed in 2.2.0

**Comment by djhoese** [comment]
Even though I told you to file this bug, I realized today I already fixed this and it has been released a couple months ago as pycoast 1.2.3.

As mentioned on slack by you, upgrading to 1.2.3 fixed your issue.
